File: Authorization.Application/DTOs/Resource/ResourceMetadataDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Authorization.Application.DTOs.Resource
{
    public record FieldDto(string Name,string BaseType, string ClrType, string? DisplayName);

    public record JoinDto(
        string NavigationName,
        string TargetEntity,
        string TargetKey,      // PK property names on the target entity
        string CurrentKey,     // FK property names on the current entity

        List<FieldDto> TargetScalarFields
    );

    public record ResourceMetadataDto(
        string ResourceKey,
        string EntityName,
        List<FieldDto> ScalarFields,
        List<JoinDto> Joins
    );

}

File: FrontEnd/src/modules/Authorization/models/ResourceMetadataDto.ts

export interface fieldDto {
  name: string;
  baseType: number;
  clrType: string;
  displayName: string;
  
} 
export interface joinDto {
  navigationName: string;
  targetEntity: string;
  targetKey: string;
  currentKey: string;
  targetScalarFields: fieldDto[];
}


export interface resourceMetadataDto {
     resourceKey: string;
     entityName: string;
      useDynamicFilter:boolean;
      useNavigate:boolean;
      useScope:boolean;
     scalarFields: fieldDto[];
     joins: joinDto[];
} 